scripts/multi_reg_and_predict_new_data.py

```diff
@@ -0,0 +1,163 @@
+            # Open CSV reader with blank separators and numeric conversion
+            print("# Reading CSV file:", file_name)
+            csv_reader = csv.reader(csv_file,
+                                    delimiter=' ',
+                                    quoting=csv.QUOTE_NONNUMERIC)
+            # Iterate over file rows
+            for row in csv_reader:
+                # Compute record type from Boolean field and update as needed
+                r_type = sum([int(b) << i for i, b in enumerate(row) if i in bool_cols])
+                if r_type not in X:
+                    # Handle particular case of first row
+                    if first_row:
+                        # Initialize number of regressors with record
+                        n_regressors = len(row) - len(excluded)
+                        print("# Found", n_regressors, "regressors in first row")
+                        first_row = False
+                        for i, x in enumerate(row):
+                            if i in excluded:
+                                continue
+                            regressor_list.append(i)
+                        if n_regressors != len(regressor_list):
+                            print("** ERROR: Unexpected regressor list length")
+                            sys.exit(1)
+
+                    # Initialize type cardinality
+                    n_obs[r_type] = 1
+                    print("# New record Boolen type found:",
+                          ' '.join([str(int(b)) for i, b in enumerate(row) if i in bool_cols]),
+                          "=>",
+                          r_type)
+                    X[r_type] = [[x] for i, x in enumerate(row) if i not in excluded]
+                else:
+                    # Increement tupe cardinality
+                    n_obs[r_type] += 1
+
+                    # Ensure consistency in number of regressors
+                    if n_regressors != len(row) - len(excluded):
+                        print("** ERROR: Incorrect number of regressors read:",
+                          len(row) - len(excluded),
+                              "!=",
+                              n_regressors)
+                        sys.exit(1)
+
+                    # Update regressors with current record
+                    j = 0
+                    for i, x in enumerate(row):
+                        if i in excluded:
+                            continue
+                        X[r_type][j].append(x)
+                        j += 1
+        
+                # Update regressand with current record
+                Y.setdefault(r_type, []).append(row[y_col])
+                ranks.setdefault(r_type, []).append(int(row[rank_col]))
+
+
+print(sys.argv)
+
+if len(sys.argv) != 2:
+    print("** ERROR: Specify training directory on the command-line")
+    sys.exit(1)
+
+training_dir = sys.argv[1];
+
+# Initialize global variables
+regressor_list = []
+n_obs = {}
```

Comment:
`{}` could be set or dict. 
Empty set could be instantiated as `set()` and empty dict as `dict()`, that would be more explicit.
I would help with these two scripts. I just need some data, to be sure, to proceed correctly :)